File: user-service/src/main/java/rs/edu/raf/repository/ClientRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.user.Client;

import java.util.List;
import java.util.Optional;

@Repository
public interface ClientRepository extends JpaRepository<Client, Long> {

    Optional<Client> findByEmailAndActiveIsTrue(String email);

    Optional<Client> findByPhoneNumberAndActiveIsTrue(String phoneNumber);

    Optional<Client> findByJMBGAndActiveIsTrue(String jmbg);

    List<Client> findAllByActiveIsTrue();

    Optional<Client> findByIdAndActiveIsTrue(Long id);
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/repository/ClientRepository.kt
package org.example.userservicekotlin.repository

import org.example.userservicekotlin.model.user.Client
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ClientRepository : JpaRepository<Client, Long> {
    fun findByEmailAndActiveIsTrue(email: String): Optional<Client>

    fun findByPhoneNumberAndActiveIsTrue(phoneNumber: String): Optional<Client>

    fun findByJMBGAndActiveIsTrue(jmbg: String): Optional<Client>

    fun findAllByActiveIsTrue(): List<Client>

    fun findByIdAndActiveIsTrue(id: Long): Optional<Client>
}